File: C++/easy/perfect-numbers/perfect_numbers.cpp
#include "perfect_numbers.h"

namespace perfect_numbers {
  perfect_numbers classify(const int& num) {
    if (num <= 0) throw std::domain_error("A domain error has occurred");
    int sum_factors{0};
    for (int i{1}; i <= num / 2; ++i) if (num % i == 0) sum_factors += i;

    if (sum_factors < num) return deficient;
    else if (sum_factors > num) return abundant;
    else return perfect;
  }
}  // namespace perfect_numbers

File: Python/easy/perfect-numbers/perfect_numbers.py
def classify(number):
    """ A perfect number equals the sum of its positive divisors.

    :param number: int a positive integer
    :return: str the classification of the input integer
    """

    if number < 1:
        raise ValueError("Classification is only possible for positive integers.")
    accumulator = 0
    for divisor in range(number // 2):
        if number % (divisor + 1) == 0:
            accumulator += divisor + 1
    if accumulator < number:
        return "deficient"
    if accumulator > number:
        return "abundant"
    return "perfect"
